### ISSUE TITLE
[Bug]: CLI config error-catching

### ISSUE THREAD

**Issue by CodyCBakerPhD** [NeurodataWithoutBorders/nwbinspector#108]
### What happened?

Also observing a very bizarre behavior where if there are actual Errors (like a `ValueError`) triggered within one of the test functions (see PR #106 that fixes the ones I'm seeing), they aren't grabbed within the `try/except` of `run_checks` and included in the report, but instead are raised to console as true errors:

![image](https://user-images.githubusercontent.com/51133164/158451334-712a6ba7-93a8-4ab0-a258-94626036bc66.png)

but if I don't include the `--config dandi` they behave as expected and go into that extra section of the final report.

### Steps to Reproduce

```shell
From a console with working directory located in a folder that has dandiset 000004 (or ros3, alternatively)...


nwbinspector ./000004 --config dandi

and compare to...

nwbinspector ./000004
```

### Traceback

```shell
Screenshot in the first section; particulars of the traceback don't really matter, it's the fact that it IS a traceback pushed to the console as the standard out instead of getting grabbed by the report.
```

### Operating System

Windows

### Python Executable

Python

### Python Version

3.9

### Package Versions

[versions.txt](https://github.com/NeurodataWithoutBorders/nwbinspector/files/8262455/versions.txt)

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/NeurodataWithoutBorders/nwbinspector/blob/dev/.github/CODE_OF_CONDUCT.rst)
- [X] Have you ensured this bug was not already [reported](https://github.com/neurodatawithoutborders/nwbinspector/issues)?
- [X] To the best of your ability, have you ensured this is a bug within the code that checks the NWBFile, rather than a bug in the NWBFile reader (e.g., [PyNWB](https://github.com/NeurodataWithoutBorders/pynwb) or [MatNWB](https://github.com/NeurodataWithoutBorders/matnwb))?

**Comment by CodyCBakerPhD** [comment]
Related to `--config` usage: https://github.com/NeurodataWithoutBorders/nwbinspector/issues/216
